src/data/chats/mod.rs

```diff
@@ -225,4 +261,113 @@ impl Chats {
             }
         }
     }
+
+    // Agents
+
+    /// Registers a new MoFa server by creating a new client, automatically testing the connection
+    /// and fetching the available agents.
+    pub fn register_mofa_server(&mut self, address: String) -> MofaServerId {
+        let server_id = MofaServerId(address.clone());
+        let client = Rc::new(MofaClient::new(address.clone()));
+        
+        self.mofa_servers.insert(server_id.clone(), MofaServer {
+            address: address.clone(),
+            client,
+            connection_status: MofaServerConnectionStatus::Connecting,
+        });
+
+        self.test_mofa_server_and_fetch_agents(&address);
+        server_id
+    }
+
+    /// Removes a MoFa server from the list of available servers.
+    pub fn remove_mofa_server(&mut self, address: &str) {
+        self.mofa_servers.remove(&MofaServerId(address.to_string()));
+        self.available_agents.retain(|_, agent| agent.server_id.0 != address);
+    }
+
+    /// Retrieves the corresponding MofaClient for an agent
+    pub fn get_client_for_agent(&self, agent_id: &AgentId) -> Option<Rc<MofaClient>> {
+        self.available_agents.get(agent_id)
+            .and_then(|agent| self.mofa_servers.get(&agent.server_id))
+            .map(|server| server.client.clone())
+    }
+
+    /// Helper method for components that need a sorted vector of agents
+    pub fn get_agents_list(&self) -> Vec<MofaAgent> {
+        let mut agents: Vec<_> = self.available_agents.values().cloned().collect();
+        agents.sort_by(|a, b| a.name.cmp(&b.name));
+        agents
+    }
+
+    /// Tests the connection to a MoFa server by requesting /v1/models
+    /// The connection status is updated at the App level based on the actions dispatched
+    /// 
+    /// We perform fetching (and therefore testing) as follows:
+    /// 1. We create a channel used to communicate with a background thread that will do the http request
+    /// 2. We start that background thread and instruct it to fetch the agents from the server
+    /// 3. The thread sends a message back to the channel created here with the server response
+    /// 4. We forward that result back to the UI thread using Cx::post_action
+    /// 5. The action is handled at the App level, which tells the store to update the connection status and agent list
```

Comment:
I would be careful with long doc comments. Unless you are making a library, long comments are likely to become unmaintained. 

I think if one is curious of how this works inside, that person will look inside. So documenting step by step how it works inside adds little value. 

I would preserve the "what it does", and "how is the result reflected":

```rust
/// Tests the connection to a MoFa server by requesting /v1/models.
///
/// The connection status is updated at the App level based on the actions dispatched.
```

Reply:
Yeah I think at the time I was trying to make clear what the changes are and I really think some explaining is needed because this mechanism of fetching and updating store/UI is quite convoluted. 

> I think if one is curious of how this works inside, that person will look inside. So documenting step by step how it works inside adds little value.

I don't think this is true here because what happens and how it happens is scattered across the application and some doc could save a lot of time. In most platforms if you say `bla.request_something` you usually get a response, not a bunch of side effects. 
However I do think this fits better in some kind of higher level documentation rather than in a specific method. 
(I should have just put this as a comment on the PR, just wanted to make reviewing easier actually).

I've shortened it with your suggestion :)